lib/index.js

```diff
@@ -28,7 +28,7 @@ api.dereferenceElectorPool = async ({electorPool, ledgerNode}) => {
   for(const e of electorPool) {
     // service may be a string referencing a serviceId contained in a DID
     // Document referenced in `e.elector`
-    if(typeof e.service === 'string') {
+    if(typeof e.service === 'string' && typeof e.elector === 'string') {
```

Comment:
Why this change? Why not just use `e.id` and use that for the record lookup as well? I don't really understand the purpose of `elector` at all at this point.

Reply:
The idea *was* that electors could have DID documents that included service descriptors.  So `elector` would contain a DID.

Reply:
But the `service` property is on the object with `id` and `elector` not *in* `elector` as an object, right? That means `id` should be a DID if we want to support that.

We should have either:

```js
{
  id: "<did/whatever>",
  // need dereferencing
  service: "<service ID>"
}
```

OR

```js
{
  id: "<did/whatever>",
  service: {
    /* serviceEndpoint etc */
  }
}
```

OR

```js
{
  id: "<whatever>",
  elector: {
    id: "<did/whatever>",
    // need dereferencing
    service: "<service ID>"
  }
}
```

OR

```js
{
  id: "<whatever>",
  elector: {
    id: "<did/whatever>",
    // need dereferencing
    service: {
      /* serviceEndpoint, etc. */
    }
  }
}
```

But right now we have some weird mishmash.

Reply:
In short, we should just do the simplest two options:

```js
{
  id: "<whatever>",
  // needs dereferencing (first part of service ID is a DID)
  service: "<service ID>"
}

OR

{
  id: "<whatever>",
  // embedded case
  service: {
    /* id, serviceEndpoint etc */
  }
}
```